renderer/execute/types.py

```diff
@@ -128,31 +130,38 @@ def _as_i18n_message(self) -> I18nMessage:
-                columns_str = f"The columns {names_str}"
+            icu_args = {
+                "columns": len(self.column_names),
+                **{
+                    str(i): self.column_names[i]
+                    for i in range(0, len(self.column_names))
+                },
+            }
 
             if self.should_be_text:
                 # Convert to Text
-                return I18nMessage.TODO_i18n(
-                    f"{columns_str} must be converted to Text."
+                # i18n: The parameter {columns} will contain the total number of columns that need to be converted; you will also receive the column names as {0}, {1}, {2}, etc.
+                return I18nMessage.trans(
+                    "py.renderer.execute.types.PromptingError.WrongColumnType.as_error_message.shouldBeText",
+                    default="{ columns, plural, offset:2"
+                    " =1 {The column “{0}” must be converted to Text.}"
+                    " =2 {The columns “{0}” and “{1}” must be converted to Text.}"
+                    " one {The columns “{0}”, “{1}” and “{2}” must be converted to Text.}"
+                    " other {The columns “{0}”, “{1}” and # others must be converted to Text.}}",
+                    args=icu_args,
                 )
             else:
-                return I18nMessage.TODO_i18n(
-                    f"{columns_str} must be converted "
-                    f"from {self.found_type_name} to {self.best_wanted_type_name}."
+                icu_args["found_type"] = self.found_type
+                icu_args["best_wanted_type"] = self.best_wanted_type_id
+                # i18n: The parameter {columns} will contain the total number of columns that need to be converted; you will also receive the column names: {0}, {1}, {2}, etc. The parameters {found_type} and {best_wanted_type} will have values among "text", "number", "datetime"; however, including a (possibly empty) "other" case is mandatory.
+                return I18nMessage.trans(
+                    "py.renderer.execute.types.PromptingError.WrongColumnType.as_error_message.general",
+                    default="{ columns, plural, offset:2"
+                    " =1 {The column “{0}” must be converted from { found_type, select, text {Text} number {Numbers} datetime {Dates & Times} other {}} to {best_wanted_type, select, text {Text} number {Numbers} datetime {Dates & Times} other {}}.}"
+                    " =2 {The columns “{0}” and “{1}” must be converted from { found_type, select, text {Text} number {Numbers} datetime {Dates & Times} other {}} to {best_wanted_type, select, text {Text} number {Numbers} datetime {Dates & Times}  other{}}.}"
+                    " one {The columns “{0}”, “{1}” and “{2}” must be converted from { found_type, select, text {Text} number {Numbers} datetime {Dates & Times} other {}} to {best_wanted_type, select, text {Text} number {Numbers} datetime {Dates & Times} other{}}.}"
```

Comment:
I'd have put `=3` here. I had to read the docs on the `offset` param to figure out why `one` selects `3`.

Reply:
Indeed, `offset` is kind of peculiar; I had to experiment to end up with the correct cases. However, it's the recommended way to make messages that read like `A and B and 4 others`.

Reply:
What I mean is: I think the `one` should be `=3`. That way the `offset:2` only affects the behavior of `#`, which seems more intuitive.